src/main/topLevelPages/myDashboard/newVirtualContributorWizard/useNewVirtualContributorWizard.tsx

```diff
@@ -496,67 +500,75 @@ const useNewVirtualContributorWizard = (): useNewVirtualContributorWizardProvide
-        },
-      });
+    const virtualContributorId = createdVc?.id;
 
-      const virtualContributorId = data?.createVirtualContributor.id;
-
-      if (virtualContributorId) {
-        if (parentRoleSetId) {
-          // the VC cannot be added to the BoK community
-          // if it's not part of the parent community
-          await addVirtualContributorToRole({
-            variables: {
-              roleSetId: parentRoleSetId,
-              contributorId: virtualContributorId,
-              role: CommunityRoleType.Member,
-            },
-          });
-        }
-
-        // add the VC to the BoK community
+    if (virtualContributorId) {
+      if (parentRoleSetId) {
+        // the VC cannot be added to the BoK community
+        // if it's not part of the parent community
         await addVirtualContributorToRole({
           variables: {
-            roleSetId: roleSetId,
+            roleSetId: parentRoleSetId,
             contributorId: virtualContributorId,
             role: CommunityRoleType.Member,
           },
         });
+      }
 
-        // add vc's nameId to the cache for the TryVC dialog
-        if (data?.createVirtualContributor.nameID) {
-          addVCCreationCache(data?.createVirtualContributor.nameID);
-        }
+      // add the VC to the BoK community
+      await addVirtualContributorToRole({
+        variables: {
+          roleSetId: roleSetId,
+          contributorId: virtualContributorId,
+          role: CommunityRoleType.Member,
+        },
+      });
+
+      notify(
+        t('createVirtualContributorWizard.createdVirtualContributor.successMessage', { name: values.name }),
+        'success'
+      );
+
+      return true;
+    }
+
+    return false;
+  };
 
-        notify(
-          t('createVirtualContributorWizard.createdVirtualContributor.successMessage', { name: values.name }),
-          'success'
-        );
+  const executeMutation = async ({ values, accountId }: { values: VirtualContributorFromProps; accountId: string }) => {
+    try {
+      const variables: CreateVirtualContributorOnAccountMutationVariables = {
+        virtualContributorData: {
+          accountID: accountId,
+          profileData: {
+            displayName: values.name,
+            tagline: values.tagline,
+            description:
+              values.description ?? t('createVirtualContributorWizard.createdVirtualContributor.description'),
+          },
+          aiPersona: {
+            aiPersonaService: {
+              engine: values.engine,
+            },
+          },
+        },
+      };
 
-        return true;
+      if (values.externalConfig) {
+        variables.virtualContributorData.aiPersona.aiPersonaService!.externalConfig = values.externalConfig;
       }
+      const { data } = await createVirtualContributor({
+        variables,
+      });
 
-      return false;
+      if (data?.createVirtualContributor.nameID) {
+        addVCCreationCache(data.createVirtualContributor.nameID);
```

Comment:
We're adding the vc to the 'cache' for the TryYourVC dialog.
We're not adding the VC to a community in this creation so you can't play with it. Let's skip  `addVCCreationCache`.

Reply:
I moved it to the expert VC handler. II thought it had to do with Apollo Cache. 👍🏻